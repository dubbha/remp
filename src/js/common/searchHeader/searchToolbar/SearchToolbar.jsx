import React from 'react';
import PropTypes from 'prop-types';
import SearchBySwitch from './SearchBySwitch';
import SearchButton from './SearchButton';

const SearchToolbar = ({ searchBy, onSearchByChange, searchByParams }) => (
  <div className="searchToolbar">
    <SearchBySwitch
      searchBy={searchBy}
      onSearchByChange={onSearchByChange}
      searchByParams={searchByParams}
    />
    <SearchButton
      text="search"
      type="submit"
      size="large"
      active
    />
  </div>
);

SearchToolbar.propTypes = {
  searchBy: PropTypes.string.isRequired,
  onSearchByChange: PropTypes.func.isRequired,
  searchByParams: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchToolbar;
