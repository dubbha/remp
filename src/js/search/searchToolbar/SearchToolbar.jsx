import React from 'react';
import SearchBySwitch from './SearchBySwitch';
import SearchButton from './SearchButton';

const SearchToolbar = () => (
  <div className="searchToolbar">
    <SearchBySwitch />
    <SearchButton
      text="search"
      type="submit"
      size="large"
      active
    />
  </div>
);

export default SearchToolbar;
