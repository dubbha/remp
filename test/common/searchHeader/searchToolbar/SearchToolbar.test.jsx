import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchToolbar from 'common/searchHeader/searchToolbar';

describe('SearchToolbar', () => {
  let props;

  beforeEach(() => {
    props = {
      searchBy: 'title',
      searchByParams: ['title', 'director'],
      onSearchByChange: jest.fn(),
    };
  });

  it('should render successfully', () => {
    const wrapper = shallow(<SearchToolbar {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
