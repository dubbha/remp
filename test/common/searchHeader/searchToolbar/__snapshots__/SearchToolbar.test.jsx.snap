// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchToolbar should render successfully 1`] = `
<div
  className="searchToolbar"
>
  <SearchBySwitch
    onSearchByChange={[Function]}
    searchBy="title"
    searchByParams={
      Array [
        "title",
        "director",
      ]
    }
  />
  <SearchButton
    active={true}
    onClickWithText={null}
    size="large"
    text="search"
    type="submit"
  />
</div>
`;
